rezasm-source/rezasm-core/src/simulation/reader.rs

```diff
@@ -0,0 +1,56 @@
+use crate::util::as_any::AsAny;
+use std::any::Any;
+use std::fmt::Debug;
+use std::io;
+
+/// A trait for any readers used with EzASM
+pub trait Reader: io::Read + AsAny + Sync + Send + Debug {
+    /// Appends a string to the reader's buffer
+    ///
+    /// # Arguments
+    ///
+    /// * new_input - input to add to this reader's buffer
+    ///
+    /// # Important
+    ///
+    /// DO NOT CALL THIS FUNCTION IN CORE.
+    /// It depends on the buffer NOT being stdin.
+    fn expand_buffer(&mut self, new_input: &str);
```

Comment:
I suspect this might not be the best way of doing this. consider trying to downcast

Reply:
@WyattRoss this one's you